### PR TITLE
add config.kit.endpointExtensions option

### PR DESCRIPTION
**Pull request by Rich-Harris**
Alternative to #4185 — prevents files other than `.js` and `.ts` files from being treated as endpoints, unless you specify `endpointExtensions`

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [x] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [x] This message body should clearly illustrate what problems it solves.
- [x] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [x] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [x] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpx changeset` and following the prompts. All changesets should be `patch` until SvelteKit 1.0

### CODE REVIEW

**Comment by Conduitry** [comment]
I do feel a bit better about this than the other option. Do we want to rename the existing `extensions` option to `pageExtensions` or something similar? Or move both under a single object?

**Comment by Rich-Harris** [comment]
The wrinkle there is that `config.extensions` is used by other tooling, like mdsvex

**Comment by Conduitry** [comment]
Oh. Hm. And it lives outside of `kit`. Never mind, I don't have a good suggestion about this then.

File: documentation/docs/13-configuration.md [hunk 1]
```diff
@@ -135,6 +136,10 @@ When pages are prerendered, the CSP header is added via a `<meta http-equiv>` ta
 
 > When `mode` is `'auto'`, SvelteKit will use nonces for dynamically rendered pages and hashes for prerendered pages. Using nonces with prerendered pages is insecure and therefore forbidden.
 
+### endpointExtensions
+
+An array of valid file extensions for endpoint modules.
```
Comment (Conduitry) [inline documentation/docs/13-configuration.md:141]: "Valid" seems like a strange word to be using here to me. It's not like Kit will throw an error for other ones, will it? Just ignore them. I don't have another word to suggest at the moment.

Comment (Rich-Harris) [reply documentation/docs/13-configuration.md:141]:
```suggestion
An array of file extensions that SvelteKit will treat as endpoints. Files with extensions that match neither `config.extensions` nor `config.kit.endpointExtensions` will be ignored by the router.
```

**Review by benmccann — APPROVED** [review]

**Pull request merged**